GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.UploadObject.cs

```diff
@@ -0,0 +1,197 @@
+        /// <returns>The metadata for the uploaded object.</returns>
+        public Object UploadObject(
+            Object destination,
+            Stream source,
+            UploadObjectOptions options,
+            IProgress<IUploadProgress> progress)
+        {
+            ValidateObject(destination);
+            source.CheckNotNull(nameof(source));
+            var mediaUpload = Service.Objects.Insert(destination, destination.Bucket, source, destination.ContentType);
+            options?.ModifyMediaUpload(mediaUpload);
+            if (progress != null)
+            {
+                mediaUpload.ProgressChanged += progress.Report;
+            }
+            var finalProgress = mediaUpload.Upload();
+            if (finalProgress.Exception != null)
+            {
+                throw finalProgress.Exception;
+            }
+            return mediaUpload.ResponseBody;
+        }
+
+        /// <summary>
+        /// Uploads the data for an object in storage asynchronously, from a specified stream.
+        /// </summary>
+        /// <param name="destination">Object to create or update. Must not be null, and must have the name and bucket populated.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <param name="options">Additional options for the upload. May be null, in which case appropriate
+        /// defaults will be used.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <param name="progress">Progress reporter for the upload. May be null.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public async Task<Object> UploadObjectAsync(
+            Object destination,
+            Stream source,
+            UploadObjectOptions options,
+            CancellationToken cancellationToken,
+            IProgress<IUploadProgress> progress)
+        {
+            ValidateObject(destination);
+            source.CheckNotNull(nameof(source));
+            var mediaUpload = Service.Objects.Insert(destination, destination.Bucket, source, destination.ContentType);
+            options?.ModifyMediaUpload(mediaUpload);
+            if (progress != null)
+            {
+                mediaUpload.ProgressChanged += progress.Report;
+            }
+            var finalProgress = await mediaUpload.UploadAsync(cancellationToken).ConfigureAwait(false);
+            if (finalProgress.Exception != null)
+            {
+                throw finalProgress.Exception;
+            }
+            return mediaUpload.ResponseBody;
+        }
+
+        private void ValidateObject(Object destination)
+        {
+            destination.CheckNotNull(nameof(destination));
+            Preconditions.CheckArgument(
+                destination.Name != null && destination.Bucket != null && destination.ContentType != null,
```

Comment:
Ok, we do check for the `ContentType` when validating the object, is just a matter of updating the comments.